src/app/component/user/components/profile/users-friends/friend-dashboard/friend-dashboard.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { LocalStorageService } from '@global-service/localstorage/local-storage.service';
+import { TranslateService } from '@ngx-translate/core';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-friend-dashboard',
+  templateUrl: './friend-dashboard.component.html',
+  styleUrls: ['./friend-dashboard.component.scss']
+})
+export class FriendDashboardComponent implements OnInit, OnDestroy {
+  public userId: number;
+  public langChangeSub: Subscription;
+
+  constructor(private localStorageService: LocalStorageService,
+              private translate: TranslateService) { }
+
+  ngOnInit() {
+    this.initUser();
+    this.subscribeToLangChange();
+    this.bindLang(this.localStorageService.getCurrentLanguage());
+  }
+
+  public initUser(): void {
+    this.localStorageService.userIdBehaviourSubject
+      .subscribe((userId: number) => this.userId = userId);
+  }
+
+  private bindLang(lang: string): void {
+    this.translate.setDefaultLang(lang);
+  }
+
+  private subscribeToLangChange(): void {
+    this.langChangeSub = this.localStorageService.languageSubject
+      .subscribe(this.bindLang.bind(this));
+  }
+
+  ngOnDestroy(): void {
+    this.langChangeSub.unsubscribe();
```

Comment:
You have 2 subscriptions and only one unsubscribe method.

Reply:
fixed